Feature/71-query-builder-error-messages

juliancoleman opened the pull request:
- fixed an issue where the `scrollable` mixin wouldn't actually apply to the height of the element applied to. Fixed by adding a height to the mixin
- displays a helpful error message if the error property exists on the executed query
- displays a helpful info message if the query returned no results
- styles for the new messages

closes #71

Commit:
fix scrollable mixin not extending the full element height

Commit:
added an error and info vm to display helpful messages

Commit:
add vm for error and info to view to display helpful error messages

Commit:
add class to element to avoid styling all h2 elements

Commit:
add scrollable mixin to record-browser

Commit:
add styles to help-text class

Commit:
refactor(record-browser): rename `err` to `error`

"Err" is a verb. Variable names should be nouns.

Commit:
refactor(record-browser): normalize how `.help-text` styles are applied

Use `.help-text.info` instead of `.help-text .info` to match the format of `.help-text.error`.
Move the default help-text color from the `h1` to the `.help-text` class to allow the `.help-text.info` color
to have an affect without placing the `color` rule on the `h1`.

Commit:
refactor(record-browser): combine `.help-text` styles

Commit:
refactor(record-browser): use semantic color variable

Commit:
feat(record-browser): adjust error message styles

Commit:
feat(varabiles): change the color of `$red` to match Material Designs's "Red 500"

https://material.google.com/style/color.html#color-color-palette

Commit:
refactor(record-browser): rename `res` to `queryResults`

Avoid abbreviations.

nick-herrera commented:
Looks good. Please include the issue number in the branch name in the future (`feature/71-query-builder-error-messages`). Also, use the karma commit message format for the PR's title. It get's reused when I merge. I added a few commits for you to review. The commit message body has some extra info in a few of them.

Before I merge, I want to check that `height: inherit;` is necessary in the `scrollable` mixin. It works fine in my browser without it. However, the mixin was applied to a custom element (`query-builder`). So I can imagine that a default `height` might not come with the user agent styles of your browser. If that's the case, I'd rather use `height: auto` instead since the height can be calculated from the `query-builder` element because it's positioned absolutely. I'm more concerned about the current element's height rather than its parent's.

We'll look at it more when I get into the office.

Commit:
refactor(mixins): remove `height: inherit;` from `scrollable` mixin

- It is unnecessary (the code works fine without it).
- If the mixin is included after a height is already defined, the original height will be overridden.
